### PR TITLE
Show a warning when navigating to unversioned urls for scripts without a course version

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Finishes [PLAT-1134](https://codedotorg.atlassian.net/browse/PLAT-1134). We support links to scripts within unit groups based on the family name of those scripts. For example, /s/csd3 redirects to /s/csd3-2019 However, for scripts within unit groups, the family name is not editable. This feature is also only supported through the 2019 curriculum and we have no plans to support this in the future.

This PR adds a dialog on those links to warn users that we will be removing them next summer:
Screen Shot 2021-09-13 at 7 29 31 AM

This is added to both the script overview page and individual script levels.

Script page showing the dialog, as well as the fact that it is not shown on the correct urls or for links using family names for standalone units.

https://user-images.githubusercontent.com/46464143/133103985-3fe55df4-20cc-4859-ad04-77adfaf3b1af.mov

The same kind of thing for script levels:

https://user-images.githubusercontent.com/46464143/133103781-1c60c190-8592-4bbf-89ee-69e25738d66c.mov

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/components/progress/UnitOverview.jsx [hunk 1]
```diff
@@ -150,10 +153,15 @@ class UnitOverview extends React.Component {
       !!scriptId &&
       isScriptHiddenForSection(hiddenLessonState, selectedSectionId, scriptId);
 
+    const showUnversionedRedirectWarningDialog =
+      showUnversionedRedirectWarning && !this.state.showRedirectDialog;
```
Comment (davidsbailey) [inline apps/src/code-studio/components/progress/UnitOverview.jsx:157]: Could you add comments here or in UnversionedScriptRedirectDialog to help distinguish this from legit redirects from urls like /s/coursea or /courses/csp ?

**Review by davidsbailey — COMMENTED** [review]
Code changes look good! My UX feedback is that I'm not super confident a teacher would know what action to take in response to the prompt:

Screen Shot 2021-09-13 at 11 28 49 AM

if a teacher just clicked a button in an LMS and then saw this, I doubt they would remember that they had come from a URL like /s/csp4. Could we perhaps ~show them the old and new urls~ give them a stronger hint? e.g.:

> You've been redirected here from a link to a unit of CSP and CSD unit which does not have a version year (e.g. /s/csp2 or /s/csd3). These links will be going away in June 2022. Please update your bookmarks to match the url you currently see in the address bar.

I'm trying to suggest something that won't require you to do more work, such as to make the dialog determine the url you redirected from, or to reimplement this as a interstitial page before redirecting (a more common practice when updating old links). 

Thoughts @tess323 ?

**Comment by tess323** [comment]
@davidsbailey @bethanyaconnor  I like that suggestion and agree we don't want to do extra work here beyond updating copy.  Tweaked the first sentence a bit - lets go with

"You've been redirected here from a link to a CSP or CSD unit that does not have a version year (e.g. /s/csp2 or /s/csd3). These links will be going away in June 2022. Please update your bookmarks to match the url you currently see in the address bar."

File: dashboard/test/ui/features/learning_platform/course_versions.feature [hunk 2]
```diff
@@ -130,6 +130,7 @@ Scenario: Switch versions using dropdown on script overview page
 Scenario: Course unit family names redirect to 2019 version
   When I am on "http://studio.code.org/s/csp3"
   And I get redirected to "/s/csp3-2019" via "dashboard"
+  And I wait until element "#ui-test-unversioned-script-redirect-dialog" is visible
```
Comment (davidsbailey) [inline dashboard/test/ui/features/learning_platform/course_versions.feature:133]: nice touch!

**Review by davidsbailey — APPROVED** [review]
Looks good!

**Pull request merged**